packages/core/src/react-integration/provider/BackupBoundary.tsx

```diff
@@ -0,0 +1,36 @@
+import React, { Suspense } from 'react';
+import { useEffect } from 'react';
+
+export default function BackupBoundary({
+  children,
+}: {
+  children: React.ReactNode;
+}) {
+  return <Suspense fallback={<Loading />}>{children}</Suspense>;
+}
+
+function Loading() {
+  let message: React.ReactNode = 'loading...';
+  /* istanbul ignore else */
+  if (process.env.NODE_ENV !== 'production') {
+    // env should not change during runtime and this excludes from build
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    useEffect(() => {
+      console.warn(
+        `Uncaught suspense.
+Make sure to add your own Suspense boundaries: https://resthooks.io/docs/getting-started/data-dependency#async-fallbacks`,
+      );
+    }, []);
```

Comment:
Yes, that's what I meant. in the case where a server render results in the error but a client render succeeds, how would it be seen without a server log?

Reply:
This a setup helper. So presumably if it could fail on server it could fail on client. If it never fails on client then it would be caught by the html sent

Reply:
Right but won’t it be user code that causes this fallback to be hit?

Reply:
This is to warn user in case they don't follow https://resthooks.io/docs/getting-started/data-dependency#async-fallbacks

It is not runtime based, but setup based and simply an aid to help new users.

Reply:
switched to useMemo I guess *shrug*